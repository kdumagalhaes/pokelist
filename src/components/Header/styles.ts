import styled from 'styled-components'

export const Wrapper = styled.header`
  background-color: ${({ theme }) => theme.red};

  height: 180px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  .container {
    display: flex;
    align-items: center;
    height: 100%;

    h1 {
      font-size: 1px;
      text-decoration: none;
      visibility: hidden;
    }
  }

  .logo {
    width: 297px;
    height: 35px;
  }

  .menu {
    background-color: ${({ theme }) => theme.black};
    padding: 0.8rem 0.5rem;
    display: flex;
    justify-content: center;
    width: 100%;
  }

  .favorites-page-link {
    background-color: ${({ theme }) => theme.darkestBlue};
    padding: 0.5rem 1rem;
    outline: 1px solid ${({ theme }) => theme.darkYellow};
    border-radius: 3px;
    color: ${({ theme }) => theme.yellow};
    text-decoration: none;
    transition: color 0.3s;
    display: flex;
    align-items: center;
    gap: 5px;

    &:hover {
      color: ${({ theme }) => theme.darkYellow};
    }
  }

  .star-icon {
    width: 20px;
    height: 20px;
  }
`
