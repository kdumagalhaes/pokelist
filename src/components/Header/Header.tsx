import { Wrapper } from './styles'
import LogoImg from '../../assets/images/logo.svg'
import StarIcon from '../../assets/icons/star-pokemon.webp'

export function Header() {
  return (
    <Wrapper>
      <div className="container">
        <a href="/">
          <img src={LogoImg} alt="" className="logo" />
        </a>
      </div>

      <nav className="menu">
        <a href="/favorites" className="favorites-page-link">
          <img src={StarIcon} alt="fav pokémon" className="star-icon" />
          My fav Pokèmons!
        </a>
      </nav>
    </Wrapper>
  )
}
