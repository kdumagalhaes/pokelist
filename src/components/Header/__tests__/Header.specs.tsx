import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Header } from '../Header'

describe('Header', () => {
  it('initial component should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render the star icon', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    const starIcon = screen.getByAltText(/fav pokémon/i)
    expect(starIcon).toBeVisible()
  })

  it('should render the logo image', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    const logo = screen.getByAltText(/logo pokelist/i)
    expect(logo).toBeVisible()
  })

  it('should render favorite text link', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    const favoriteText = /my favorite pokémon/i
    expect(screen.getByText(favoriteText)).toBeInTheDocument()
  })

  it('should navigate to favorites page', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    const path = '/favorites'
    const linkText = /my favorite pokémons/i
    expect(screen.getByText(linkText)).toHaveAttribute('href', path)
  })
})
