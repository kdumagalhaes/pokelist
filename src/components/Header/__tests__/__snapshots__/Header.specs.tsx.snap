// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header initial component should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="sc-bcXHqe bXuyeb"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <h1>
          logo poketlist
        </h1>
        <img
          alt="logo pokelist"
          class="logo"
          src="logo.svg"
        />
      </a>
    </div>
    <nav
      class="menu"
    >
      <a
        class="favorites-page-link"
        href="/favorites"
      >
        <img
          alt="fav pokémon"
          class="star-icon"
          src="star-pokemon.webp"
        />
        My favorite Pokémons!
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
